data/test/macros/wml_unit_test_macros.cfg

```diff
@@ -171,9 +171,23 @@ bob#endarg
 #arg COMPARE
 numerical_equals#endarg
 
+#arg DEATH_RESULT
+FAIL#endarg
+
     [event]
         name = start
 
+        # allow specifying a different number of strikes per weapon
+        [set_variables]
+            name=strikes
+            [split]
+                list={STRIKES}
+                key=value
+                separator=","
+                remove_empty=yes
+            [/split]
+        [/set_variables]
```

Comment:
Would be good to assert that the length of the expectations matches the number of weapons that the unit has

Reply:
How do I check the number of weapons a unit has?

Reply:
Using `{ID1}` as the variable name seems to be the style for this file, which is why I'm doing it too:
```
        [store_unit]
            [filter]
                id={ID1},{ID2}
            [/filter]
            variable={ID1}
        [/store_unit]

        {ASSERT ({VARIABLE_CONDITIONAL {ID1}.attack.length numerical_equals $strikes.length})}
        {CLEAR_VARIABLE {ID1}}
```